### DISCUSSION TITLE
Proposal: Namespace methods according to respective Proto file

### DISCUSSION THREAD

**Discussion by justin-lovell** [DCS-gRPC/rust-server#127] [Ideas]
In browsing the source code, I may have generated a conflict without realizing it at the time. Here are two functions which both define `getPlayers`:

https://github.com/DCS-gRPC/rust-server/blob/b1d39063f844f176ef619185db99200f6f1f3d43/lua/DCS-gRPC/methods/coalitions.lua#L153-L160

https://github.com/DCS-gRPC/rust-server/blob/c437fb4d80c2070dc713d93bf135722b864ac7d1/lua/DCS-gRPC/methods/net.lua#L24-L43

I would like to propose to avoid naming conflicting, let's prefix the method name with some annotation of the proto file as a namespace.

E.g. 
```diff
- GRPC.methods.getPlayers = function(params)
+ GRPC.methods.net_getPlayers = function(params)
```
and
```diff
- GRPC.methods.getPlayers = function(params)
+ GRPC.methods.coalition_getPlayers = function(params)
```

An alternative is setup a localized Lua table with the following pseudo:
```lua
local coalition = {}
GRPC.methods.coalition = coalition

coalition.getPlayers = function() {
-- ...
}
```
and the Rust code would perform a function call similar to:
```diff
- self.request("getPlayers", request)
+self.request("coalition", "getPlayers", request)
```

**Comment by rurounijones** [comment]
Generally I would go with the simplest thing and just prefix like `net_getPlayers`.

However in this _particular_ case I wonder if we shouldn't instead change the API names to be a bit more specifically accurate. `coalition.getPlayers` was named as such because that was the ED name.

However, especially in light of the conflict above, I wonder if we should rename this to `coalition.getPlayerUnits` since it is actually a more accurate representation of the data being returned which is in contrast to the data being returned by `net.getPlayers` which is more about the player than the unit.

**Comment by justin-lovell** [comment]
> I wonder if we should rename this to `coalition.getPlayerUnits` since it is actually a more accurate representation of the data being returned which is in contrast to the data being returned by `net.getPlayers` which is more about the player than the unit.

Good observation. I will integrate that in a PR to come shortly.

This morning, I thought of a lightweight solution to detect conflicts in registration. What if we had functional registrations such as
```lua
GRPC.registerMethod('theMethodName', function(params)

end)
```

If there is already another method, deregister both methods. That way the person who is writing a new function  would be forced to acknowledge the duplication.

The only other alternative is to perform a full regression test suite, which is very heavy given for full confidence, we need to launch DCS on a CI build (ouch!)
